### ISSUE TITLE
LS- GAN code is missing.

### ISSUE THREAD

**Issue by Ishaan2605** [Prarabdha14/GAN-Comparison#1]

**Comment by Kulkarni-ui** [comment]
please find the attached code 

        # Training Loop for LS-GAN
         num_epochs = 50
        for epoch in range(num_epochs):
        for i, (real_imgs, _) in enumerate(dataloader):
        real_imgs = real_imgs.to(device)
        batch_size = real_imgs.size(0)
        noise = torch.randn(batch_size, 100, device=device)
        fake_imgs = generator(noise)

        # Train Discriminator
        optimizer_D.zero_grad()
        real_out = discriminator(real_imgs)
        fake_out = discriminator(fake_imgs.detach())
        loss_D = criterion(real_out, torch.ones_like(real_out)) + criterion(fake_out, torch.zeros_like(fake_out))
        loss_D.backward()
        optimizer_D.step()

        # Train Generator
        optimizer_G.zero_grad()
        fake_out = discriminator(fake_imgs)
        loss_G = criterion(fake_out, torch.ones_like(fake_out))
        loss_G.backward()
        optimizer_G.step()

    print(f"Epoch [{epoch+1}/{num_epochs}] | Loss D: {loss_D.item():.4f} | Loss G: {loss_G.item():.4f}")

**Comment by kanikathombre** [comment]
def evaluate_lsgan(generator, z_dim, num_samples=100):
    generator.eval()
    fake_images = [ ]

    with torch.no_grad():
        for _ in range(num_samples // 64):
            noise = torch.randn(64, z_dim, device=device)
            generated_batch = generator(noise).cpu()  # Move to CPU
            generated_batch = ((generated_batch + 1) * 127.5).clamp(0, 255).byte()  # Convert to uint8

            # Convert 1-channel grayscale to 3-channel RGB for Inception Model
            generated_batch = generated_batch.repeat(1, 3, 1, 1)

            fake_images.append(generated_batch)

    fake_images = torch.cat(fake_images, dim=0)

    # Inception Score
    is_metric = InceptionScore()
    is_score = is_metric(fake_images)

    # FID Score
    fid_metric = FrechetInceptionDistance(normalize=True)

    # Get real images from MedMNIST
    real_images, _ = next(iter(train_loader))  # Use train_loader instead of dataloader
    real_images = real_images[:num_samples].cpu()  # Move to CPU
    real_images = ((real_images + 1) * 127.5).clamp(0, 255).byte()  # Convert to uint8

    # Convert real images to 3-channel RGB
    real_images = real_images.repeat(1, 3, 1, 1)

    # Compute FID
    fid_metric.update(real_images, real=True)
    fid_metric.update(fake_images, real=False)
    fid_value = fid_metric.compute()

    print(f"Inception Score (IS): {is_score[0].item():.4f} ± {is_score[1].item():.4f}")
    print(f"Fréchet Inception Distance (FID): {fid_value.item():.4f}")

# Run evaluation
evaluate_lsgan(generator, z_dim)
